### PR TITLE
Remove unused route resource properties

### PR DESCRIPTION
**Pull request by tas50**
Add descriptions and remove unused properties.

### CODE REVIEW

File: lib/chef/provider/route.rb [hunk 1]
```diff
@@ -154,58 +154,57 @@ def action_delete
         end
 
         # for now we always write the file (ugly but its what it is)
-        generate_config
+        write_config
       end
 
-      def generate_config
+      def write_config
```
Comment (lamont-granquist) [inline lib/chef/provider/route.rb]: this could be "public API" if anyone subclasses.

File: lib/chef/provider/route.rb [hunk 2]
```diff
@@ -154,58 +154,57 @@ def action_delete
...
+        return unless platform_family?("rhel", "fedora", "amazon")
+
         conf = {}
-        case node[:platform]
-        when "centos", "redhat", "fedora"
```
Comment (lamont-granquist) [inline lib/chef/provider/route.rb]: kind of like the case statement here better than the early return, its more indicative that this is where you'd need to add support for debian and other linuxes...

File: lib/chef/resource/route.rb [hunk 3]
```diff
@@ -27,20 +27,26 @@ class Route < Chef::Resource
...
-      property :gateway, [String, nil]
-      property :device, [String, nil], desired_state: false # Has a partial default in the provider of eth0.
-      property :route_type, [:host, :net], default: :host, coerce: proc { |x| x.to_sym }, desired_state: false
-
-      # I can find no evidence of these properties actually being used by Chef. NK 2017-04-11
-      property :networking, [String, nil], desired_state: false
-      property :networking_ipv6, [String, nil], desired_state: false
-      property :hostname, [String, nil], desired_state: false
-      property :domainname, [String, nil], desired_state: false
-      property :domain, [String, nil], desired_state: false
```
Comment (thommay) [inline lib/chef/resource/route.rb]: These are all part of the `route` public API and can't be removed until chef 15

Comment (tas50) [reply lib/chef/resource/route.rb]: That's totally fine. This can wait till we branch off chef-14.

**Pull request merged**